components/feature/prompts/src/main/java/mozilla/components/feature/prompts/PromptFeature.kt

```diff
@@ -433,37 +458,23 @@ class PromptFeature(
         }
     }
 
-    /**
-     * Observes [Session.Observer.onPromptRequested] of the selected session
-     * and notifies the feature whenever a prompt needs to be shown.
-     */
-    internal class PromptRequestObserver(
-        sessionManager: SessionManager,
-        private val feature: PromptFeature
-    ) : SelectionAwareSessionObserver(sessionManager) {
-
-        override fun onPromptRequested(session: Session, promptRequest: PromptRequest): Boolean {
-            feature.onPromptRequested(session, promptRequest)
-            return false
-        }
-
-        override fun onLoadingStateChanged(session: Session, loading: Boolean) {
-            if (!loading) {
-                feature.promptAbuserDetector.resetJSAlertAbuseState()
-            }
-        }
-    }
-
     companion object {
         const val FILE_PICKER_ACTIVITY_REQUEST_CODE = 1234
     }
 }
 
-internal fun SessionManager.consumePromptFrom(
+internal fun BrowserStore.consumePromptFrom(
     sessionId: String?,
-    consumer: (PromptRequest) -> Boolean
+    consume: (PromptRequest) -> Unit
 ) {
-    runWithSessionIdOrSelected(sessionId) {
-        it.promptRequest.consume(consumer)
+    if (sessionId == null) {
+        state.selectedTab
```

Comment:
Note: We have to do this as `findCustomTabOrSelectedTab` is not good enough here. We're calling this method with session IDs that can be either for custom tab or regular tabs. If `findCustomTabOrSelectedTab` is called with an ID it always assumes it is for a custom tab.